### ISSUE TITLE
How does LLaVA-NeXT (llama3-llava-next-8b version) do batch inference?

### ISSUE THREAD

**Issue by SCAI00** [LLaVA-VL/LLaVA-NeXT#433]
How does LLaVA-NeXT (llama3-llava-next-8b version) do batch inference? I've tried a number of scenarios with no success. My code is as follows:
`conv_template = "llava_llama_3" # Make sure you use correct chat template for different models
  question1 = DEFAULT_IMAGE_TOKEN + f"\n Does this image contain a book? Yes or No?
  conv_1 = copy.deepcopy(conv_templates[conv_template])
  conv_1.append_message(conv_1.roles[0], question1)
  conv_1.append_message(conv_1.roles[1], None)
  conv_1.tokenizer = self.tokenizer
  prompt_question_1 = conv_1.get_prompt()
  input_ids_Q1 = tokenizer_image_token(prompt_question_1, self.tokenizer, IMAGE_TOKEN_INDEX, return_tensors="pt").unsqueeze(0).to(device)

  conv_template = "llava_llama_3" # Make sure you use correct chat template for different models
  question_2 = DEFAULT_IMAGE_TOKEN + f"\n Describe this image."
  conv_2 = copy.deepcopy(conv_templates[conv_template])
  conv_2.append_message(conv_2.roles[0], question_2)
  conv_2.append_message(conv_2.roles[1], None)
  conv_2.tokenizer = self.tokenizer
  prompt_question_2 = conv_2.get_prompt()

  input_ids_Q2 = tokenizer_image_token(prompt_question_2, self.tokenizer, IMAGE_TOKEN_INDEX, return_tensors="pt").unsqueeze(0).to(device)

  batch_input_ids = [input_ids_Q1, input_ids_Q2]   # input_ids_Q1.shape --> torch.Size([1, 63]).   input_ids_Q2.shape --> torch.Size([1, 55])
  max_len = max([len(seq.squeeze()) for seq in batch_input_ids])

  # Pad each sequence in input_ids to the max_len
  padded_input_ids = [pad_sequence_to_max_length(seq.squeeze(), max_len) for seq in batch_input_ids]
  prompts = torch.stack(padded_input_ids)   # prompts.shape --> torch.Size([2, 63])

  url1 = "https://github.com/haotian-liu/LLaVA/blob/1a91fc274d7c35a9b50b3cb29c4247ae5837ce39/images/llava_v1_5_radar.jpg?raw=true"
  url2 = "https://raw.githubusercontent.com/haotian-liu/LLaVA/main/images/llava_logo.png"
  
  image1 = Image.open(requests.get(url1, stream=True).raw)
  image2 = Image.open(requests.get(url2, stream=True).raw)
  
  images = [image1, image2]
  image_tensors = process_images(images, image_processor, model.module.config)
  image_tensors_ = [_image.to(dtype=torch.float16, device=device) for _image in image_tensors]
  image_sizes = [(640, 427), (640, 427)]
  
  cont = model.generate(
      prompts,
      images=image_tensors_,
      image_sizes=image_sizes,
      do_sample=False,
      temperature=0,
      max_new_tokens=256,
  )
  text_output = tokenizer.batch_decode(cont, skip_special_tokens=True)`

`def pad_sequence_to_max_length(sequence, max_length, padding_value=0):
    """Pad a sequence to the desired max length (right padding)."""
    sequence = sequence.squeeze()
    print('sequence.shape[0]:', sequence.shape[0])
    print('max_length:', max_length)

    if sequence.shape[0] >= max_length:
        return sequence

    padding = torch.full((max_length - sequence.shape[0],), padding_value, dtype=sequence.dtype)
    padding = padding.cuda()

    print('padding.shape:', padding.shape)
    print('max_length - sequence.shape[0]:', max_length - sequence.shape[0])
    return torch.cat([sequence, padding])`
My input contains two questions and two images, but the reply contains only one ANSWER, which is as follows:
['\nNo, the image does not contain a book. It appears to be a radar chart or a spider chart, which is a type of graph used to display multivariate data in the form of a two-dimensional chart of three or more quantitative variables represented on axes starting from the same point.']

**Comment by GOTA404** [comment]
same question
